core/src/main/java/feign/SynchronousMethodHandler.java

```diff
@@ -26,7 +26,7 @@
 import static feign.Util.checkNotNull;
 import static feign.Util.ensureClosed;
 
-final class SynchronousMethodHandler implements MethodHandler {
+public class SynchronousMethodHandler implements MethodHandler {
```

Comment:
Changing this visibility breaks the **Open-Closed** principle current held by `feign-core`.  This is a single implementation of a `MethodHandler`.  Currently, libraries that want to extend this, do so by Composition, not Inheritance.  

I understand why you did this, to support the builder, but not all Method Handlers are going to need the same things.  For instance, the HystrixMethod handler in `feign-hystrix` would not be able to conform.

Reply:
If this can't be overridden, then there is no way to change how each method is processed.  I would like to be able to use the JAX-WS classes that I have, and can't change, to be used to build a Feign client.

Reply:
What you can do instead is create your own `InvocationHandlerFactory` and then you use this method handler through a delegate.  Take a look at the `feign-hystrix` for an example.

Reply:
Here is the reference for you.  By doing this, you can wrap the method execution and do additional changes to it.

https://github.com/OpenFeign/feign/blob/92e6ad85630ca3318e9f5e4cf12ac58806bb2645/hystrix/src/main/java/feign/hystrix/HystrixInvocationHandler.java#L85-L115